File: app/assets/javascripts/channels/smoke_binpartner.js
$(function() {
  if ($('.ws_sender').attr('id') == "smoke-binpartner") {
    App.smoke_binpartner = App.cable.subscriptions.create("SmokeBinpartnerChannel", {
      connected: function() {
        $('.ws label').text("WebSocket connect").css("color", "yellow");
      },

      disconnected: function() {
        $('.ws label').text("WebSocket disconnected").css("color", "red");
      },

      received: function() {
        function_handler_start("Run Smoke test Binpartner")
        function_handler_done("Done Smoke test Binpartner")

        function_stop_stop("KILL PROCCESS TEST")

      }
    });

    $('.pid_process').click(function(){
      $('.stop').click()
      App.smoke_binpartner.send(JSON.stringify({
        pid_cucumber: "pid_cucumber"
      }))
    });

};

    $('#smoke-binpartner').click(function(){

      $(".step_cucumber").click();
      $(".start").click();
      window.scrollTo(0, 999999);

      App.smoke_binpartner.send(JSON.stringify({
        smoke_binpartner: true,  //!!!!!
        count_features: get_data(),
        thread: $('#generator_thread').val(),
        clear_cache: $('#generator_clear_clear_cache').prop('checked'),
        proxy_server: $('#generator_proxy_server').val(),
        proxy_http: "--proxy-server=" + $('#generator_proxy_http').val(),
        ip_address: App.config.ip,
        param_headless: $('#generator_param_headless').val(),

      }));

      return false;
  })
});

File: app/channels/ws/binpartner/smoke_binpartner.rb
require "./features/helpers/decor"

module Server
  class SmokeBinpartner
    attr_accessor :smoke_binpartner, :count_features, :thread, :clear_cache, :proxy_http, :proxy_server, :ip_address, :param_headless

    def initialize(params)
      @params = params
      params.each { |key, value| public_send("#{key}=", value) }
    end

    def get_link
      JSON.parse(@params.to_json).map { |key, value| key == "count_features" ? value.to_s : "#{key}=#{value.to_s}" }.join(" ")
    end

    def message
      thread.to_i.times do
        if clear_cache == "true"
          puts "Clear cache".yellow
          system "rake tmp:cache:clear"
          system "rm -rf tmp"
          puts "Cache clear done".yellow
        end

        puts
        puts get_link.yellow
        puts
        send_broadcast "smoke_binomo", "Run Smoke test Binpartner"
        system get_link
        send_broadcast "smoke_binomo", "Done Smoke test Binpartner"
      end
    end

  end
end
